### PR TITLE
swayosd: unstable-2023-09-26 -> 0-unstable-2024-04-15

### PR DESCRIPTION
**Pull request by SergioRibera**
## Description of changes
swayosd: unstable-2023-09-26 -> unstable-2024-04-15
homepage: [ErikReider/SwayOSD](https://github.com/ErikReider/SwayOSD)

**Changelog**
- New brightnessctl backend, set by percentage, and pick device
- Added server option to use custom user CSS

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
Maybe also patch brightnessctl path? https://github.com/ErikReider/SwayOSD/blob/11271760052c4a4a4057f2d287944d74e8fbdb58/src/brightness_backend/brightnessctl.rs#L63

**Comment by SergioRibera** [comment]
> Maybe also patch brightnessctl path? [ErikReider/SwayOSD@`1127176`/src/brightness_backend/brightnessctl.rs#L63](https://github.com/ErikReider/SwayOSD/blob/11271760052c4a4a4057f2d287944d74e8fbdb58/src/brightness_backend/brightnessctl.rs#L63)

The binary should be available globally from the `PATH` environment variable, I would think that in any case you have to place that package as a runtime dependency.

**Comment by Aleksanaa** [comment]
> The binary should be available globally from the `PATH` environment variable, I would think that in any case you have to place that package as a runtime dependency.

In this case we would usually override the path in the wrapper, but we have already done enough substitutions, so we can just replace it directly.

File: pkgs/by-name/sw/swayosd/package.nix [hunk 1]
```diff
@@ -12,25 +12,26 @@
...
 , rustc
+, sassc
 , stdenv
 , udev
 }:
 
 stdenv.mkDerivation rec {
   pname = "swayosd";
-  version = "unstable-2023-09-26";
+  version = "unstable-2024-04-15";
```
Comment (kirillrdy) [inline pkgs/by-name/sw/swayosd/package.nix:22]:
```suggestion
  version = "0-unstable-2024-04-15";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

Comment (SergioRibera) [reply pkgs/by-name/sw/swayosd/package.nix]: Done!

Comment (kirillrdy) [reply pkgs/by-name/sw/swayosd/package.nix]: you will also need to adjust commit message and PR title

**Comment by SergioRibera** [comment]
@Aleksanaa How about something like this instead of the patch?

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/avizo/default.nix#L38
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/nwg-panel/default.nix#L52

**Comment by Aleksanaa** [comment]
Also okay.
```
preFixup = ''
  gappsWrapperArgs+=(
    --prefix PATH : ${lib.makeBinPath [ brightnessctl ]}
  )
'';
```

**Comment by SergioRibera** [comment]
@Aleksanaa Could you please review, I just modified it and did local builds and execution.

**Comment by Aleksanaa** [comment]
Also the commit message has to be reworded

**Pull request merged**